api-operator/pkg/controller/integration/integration_deployment.go

```diff
@@ -79,13 +79,50 @@ func (r *ReconcileIntegration) deploymentForIntegration(config EIConfigNew) *app
 	if m.Spec.DeploySpec.LimitCpu != "" {
 		limit[corev1.ResourceCPU] = resource.MustParse(m.Spec.DeploySpec.LimitCpu)
 	}
-	if m.Spec.DeploySpec.MemoryLimit !="" {
+	if m.Spec.DeploySpec.MemoryLimit != "" {
 		limit[corev1.ResourceMemory] = resource.MustParse(m.Spec.DeploySpec.MemoryLimit)
 	}
 
 	livenessProbe, _ := getLivenessProbe(config)
 	readinessProbe, _ := getReadinessProbe(config)
 
+	volMounts := make([]corev1.VolumeMount, 0)
+	volumes := make([]corev1.Volume, 0)
+	for _, configMapDetail := range m.Spec.DeploySpec.ConfigMapDetails {
+		volumes = append(
+			volumes,
+			corev1.Volume{
+				Name: "volume-" + configMapDetail.Name,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: configMapDetail.Name,
+						},
+					},
+				},
+			},
+		)
+
+		volMounts = append(
+			volMounts,
+			corev1.VolumeMount{
+				Name:      "volume-" + configMapDetail.Name,
+				MountPath: configMapDetail.MountPath + "/" + configMapDetail.FileName,
+				SubPath:   configMapDetail.FileName,
+			},
+		)
+	}
+	pullPolicy := m.Spec.DeploySpec.PullPolicy
+	imagePullPolicy := corev1.PullAlways
```

Comment:
Instead of defining value like this, shall we use a `default` condition in the switch statement?

Reply:
Fixed